test/call-type-api-test.js

```diff
@@ -0,0 +1,61 @@
+const chai = require('chai');
+const assert = chai.assert;
+const request = require('request');
+const app = require('../app');
+const routes = require('../routes/index');
+
+
+describe('CallType API', () => {
+
+  before((done) => {
+    this.port = 9876;
+
+    this.server = app.listen(this.port, (err, result) => {
+      if (err) { return done(err); }
+      done();
+    });
+
+    this.request = request.defaults({
+      baseUrl: 'http://localhost:9876/'
+    });
+  });
+
+  after(() => {
+    this.server.close();
+  });
+
+  describe('GET /api/v1/call_types', () => {
+    it('should return an array of call_types', (done) => {
+      this.request.get('/api/v1/call_types', (error, response) => {
+        if (error) { done(error); }
+        var parsed = JSON.parse(response.body)
+        assert.isArray(parsed);
+        done();
+      });
+    });
+
+    it('should return 311 call_types', (done) => {
+      this.request.get('/api/v1/call_types', (error, response) => {
+        if (error) { done(error); }
+        var parsed = JSON.parse(response.body)
+        assert.equal(parsed.length, 311);
```

Comment:
Thanks Meeka - I really appreciate the comments.  I plan to go back and make all of the query string fields variables, and will do these length ones that way as well.  

So... my DB is a 1MM record DB that I created from CSV files and seriously cleaned a few weeks before this mod.  Nothing about my app will modify data in the DB - it's just querying it.  I didn't really anticipate making a separate DB for testing - other than incidents, none of the other tables will change over time (and incidents will only change in the sense that more will be added). Do you suggest that I work with a test DB given these circumstances?